File: Python/ParkingLot.py
class ParkingLot:
    def __init__(self, capacity):
        # Konstruktor untuk menginisialisasi objek ParkingLot dengan kapasitas dan daftar kendaraan kosong
        self.__capacity = capacity  # Menetapkan kapasitas maksimum dari tempat parkir
        self.__vehicles = []        # Menginisialisasi sebuah daftar kosong untuk menyimpan kendaraan yang diparkir
    
    def park_vehicle(self, vehicle):
        # Metode untuk memarkirkan kendaraan di tempat parkir
        if len(self.__vehicles) < self.__capacity:  # Memeriksa apakah ada ruang kosong di tempat parkir
            self.__vehicles.append(vehicle)        # Menambahkan kendaraan ke daftar kendaraan yang diparkir
            
            print(f"{vehicle.get_plate_number()} berhasil diparkir!")  # Mencetak pesan sukses
        else:
            print(f"{vehicle.get_plate_number()} tidak dapat parkir karena tempat parkir penuh!")         # Mencetak pesan yang menunjukkan tempat parkir penuh
    
    def display_parked_vehicles(self):
        # Metode untuk menampilkan detail kendaraan yang diparkir di tempat parkir
        
        print("Kendaraan yang diparkir di tempat parkir:")
        for vehicle in self.__vehicles:                             # Mengulangi daftar kendaraan yang diparkir
            print(f"Nomor Plat: {vehicle.get_plate_number()}\n"     # Mencetak nomor plat kendaraan
                  f"Merk      : {vehicle.get_brand()}\n"                   # Mencetak merk kendaraan
                  f"Tahun     : {vehicle.get_year()}\n"                  # Mencetak tahun kendaraan
                  f"Warna     : {vehicle.get_color()}\n")                  # Mencetak warna kendaraan

File: cpp2/ParkingLot.cpp
#include <iostream>
#include <vector>
#include "Vehicle.cpp" // Saya asumsikan file ini mengandung implementasi kelas Vehicle

class ParkingLot {
private:
    int capacity;               // Kapasitas maksimum tempat parkir
    std::vector<Vehicle> vehicles; // Vektor untuk menyimpan daftar kendaraan yang diparkir

public:
    // Konstruktor untuk menginisialisasi objek ParkingLot dengan kapasitas dan daftar kendaraan kosong
    ParkingLot(int capacity) : capacity(capacity) {}

    // Getter untuk kapasitas tempat parkir
    int get_capacity() const {
        return capacity;
    }

    // Setter untuk kapasitas tempat parkir
    void set_capacity(int new_capacity) {
        capacity = new_capacity;
    }

    // Metode untuk memarkirkan kendaraan di tempat parkir
    void park_vehicle(Vehicle vehicle) {
        if (vehicles.size() < capacity) {  // Memeriksa apakah ada ruang kosong di tempat parkir
            vehicles.push_back(vehicle);   // Menambahkan kendaraan ke daftar kendaraan yang diparkir
            std::cout << vehicle.get_plate_number() << " berhasil diparkir!" << std::endl;  // Mencetak pesan sukses
        } else {
            std::cout << vehicle.get_plate_number() << " tidak dapat parkir karena tempat parkir penuh!" << std::endl; // Mencetak pesan yang menunjukkan tempat parkir penuh
        }
    }

    // Metode untuk menampilkan detail kendaraan yang diparkir di tempat parkir
    void display_parked_vehicles() {
        std::cout << "Kendaraan yang diparkir di tempat parkir:" << std::endl;
        for (const auto& vehicle : vehicles) { // Mengulangi daftar kendaraan yang diparkir
            std::cout << "Nomor Plat: " << vehicle.get_plate_number() << std::endl;   // Mencetak nomor plat kendaraan
            std::cout << "Merk      : " << vehicle.get_brand() << std::endl;          // Mencetak merk kendaraan
            std::cout << "Tahun     : " << vehicle.get_year() << std::endl;           // Mencetak tahun kendaraan
            std::cout << "Warna     : " << vehicle.get_color() << std::endl;          // Mencetak warna kendaraan
        }
    }
};
